lib/services/cloud-codesign-service.ts

```diff
@@ -69,22 +67,26 @@ export class CloudCodesignService extends CloudService implements ICloudCodesign
 
 	private async executeGeneration(codesignData: ICodesignData,
 		projectDir: string,
-		buildId: string): Promise<ICodesignResultData> {
+		cloudOperationId: string): Promise<ICodesignResultData> {
 		const codesignInformationString = "generation of iOS certificate and provision files";
 		this.$logger.info(`Starting ${codesignInformationString}.`);
 
 		const projectData = this.$projectDataService.getProjectData(projectDir);
-		const codesignRequest = await this.prepareCodesignRequest(buildId, codesignData, projectData);
+		const codesignRequest = await this.prepareCodesignRequest(cloudOperationId, codesignData, projectData);
 		const codesignResponse: IServerResponse = await this.$nsCloudServerBuildService.generateCodesignFiles(codesignRequest);
 		this.$logger.trace(`Codesign response: ${JSON.stringify(codesignResponse)}`);
-
+		let codesignResult;
 		try {
-			await this.waitForServerOperationToFinish(buildId, codesignResponse);
+			codesignResult = await this.waitForServerOperationToFinish(cloudOperationId, codesignResponse, { silent: false });
 		} catch (ex) {
+			codesignResult = this.getResult(cloudOperationId);
```

Comment:
[Suggestion] Move this.getResult to the catch in waitForServerOperationToFinish

Reply:
Not all cloud services need the result in case of error and if we move it there, we are going to loose the error.